src/app/code/community/Hackathon/PromoCodeMessages/Model/Validator.php

```diff
@@ -170,6 +170,15 @@ protected function _validateGeneral($rule, $coupon)
             }
         }
 
+        // magemail coupon-level auto-expiration date
+        if ($coupon->getdata('magemail_expired_at') && $coupon->getUsageLimit() && $coupon->getTimesUsed() >= $coupon->getUsageLimit()) {
```

Comment:
is `getdata` correct here? if so, please make it conform to the other usage in this line, e.g. `getMagemailExpiredAt`